projects/snapshot_bot/snapshot_bot.cc

```diff
@@ -541,6 +548,7 @@ class RobotFSM : FSM<State>::StateHandler
 #ifdef USE_VICON
     // Vicon tracking interface
     Vicon::UDPClient<Vicon::ObjectData> m_ViconTracking;
+    BN005 imu;
```

Comment:
Atm it'll only record IMU data if ``USE_VICON`` is enabled, which is not a biggie, but I don't think we should have this restriction.

More importantly though, we now won't be able to use ``snapshot_bot`` *without* the IMU -- if it's not available then we'll just get an error. The way to get round this is to make it a (smart) pointer, e.g. ``std::unique_ptr<BN003> m_IMU;``. Then if the user *actually* wants to use the IMU (i.e. if they've set a value in the config file), you can create the object later, with something like:
```c++
if (m_Config.shouldUseIMU()) {
    m_IMU = std::make_unique<BN055>();
}
```

You should definitely fix the other (compile-time) issues first though, then you can make this change after.

Reply:
Hey Alex, 
I adressed all the other things, but this part does not compile. 
No matter if I do 

`BN005 imu;`
or
`m_IMU = std::make_unique<BN055>();` 

I get a "does not name a type error". Is there some header file in which I should forward declare?
Did this compile for you?

Reply:
It should be ``BN055 imu;`` I think (you've got the class name wrong).

Also, you shouldn't do the ``make_unique`` stuff where you're declaring the member variables! The whole point is that you don't necessarily want to assign a value to it, in the case that you're not using the IMU. (It's basically the equivalent of assigning ``None`` to a variable in Python.) You'd declare it as something like ``std::unique_ptr<BN055> m_IMU;`` and you later (optionally) assign to it (with something like the snippet I sent you) in the constructor depending on whether or not it's needed.